File: concrete/src/main/java/com/jaynewstrom/concrete/ConcreteWallInitializationAction.java
package com.jaynewstrom.concrete;

/**
 * Can be used when calling {@link ConcreteWall#stack} and will be called when the wall is created.
 */
public interface ConcreteWallInitializationAction<C> {
    /**
     * Called exactly once for each {@link ConcreteWall} instance.
     */
    void onWallInitialized(ConcreteWall<C> wall);
}

File: concrete/src/main/java/com/jaynewstrom/concrete/ConcreteWallInitializationAction.kt
package com.jaynewstrom.concrete

/**
 * Can be used when calling [ConcreteWall.stack] and will be called when the wall is created.
 */
interface ConcreteWallInitializationAction<in C> {
    /**
     * Called exactly once for each [ConcreteWall] instance.
     */
    fun onWallInitialized(wall: ConcreteWall<C>)
}
